### DISCUSSION TITLE
Imagick php class is missing

### DISCUSSION THREAD

**Discussion by Sommy181** [lowag/libremaint#5] [Q&A]
Imagick php class is missing 
I am facing this error

Please Help !

**Comment by Sommy181** [comment]
![image](https://user-images.githubusercontent.com/98112749/150384127-1aeaca2f-2661-41ae-ad1f-4db011de0680.png)

**Reply by lowag** [reply]
If you are using a kind of Ubuntu variant:
    sudo apt-get install php-imagick.
    sudo service apache2 restart

**Reply by Sommy181** [reply]
i am using xammp server on windows

**Reply by Sommy181** [reply]
And I am also getting php warnings and php notices of undefined variables even though I didn't edit any php files..

**Reply by lowag** [reply]
You can try this: https://shouts.dev/how-to-install-imagemagick-php-ext-on-windows , but I don't have any experience in WAMP servers
Theoreticly al warnings and notices are disabled in config/lm_settings.php ...

**Reply by Sommy181** [reply]
what about this error ?

TCPDF ERROR: [Image] Unable to get the size of the image: C:/xampp/htdocs/temp1.png

![image](https://user-images.githubusercontent.com/98112749/150398068-0e3b2e93-9b9c-43ca-898c-fd9edd8a7c75.png)

**Reply by lowag** [reply]
I guess it is on the "Queries" page when you are trying making a pdf report. Have you got data in your database? (Workorders->works) If there is please check whether C:/xampp/htdocs/temp1.png exists.

**Reply by whiteCosmic** [reply]
@lowag I followed the instructions of the installation of imagick.
Couldn't find a imagick version that is compatible with php 8.2.

on [peck.php.net](http://peck.php.net/) I could only find imagick 3.7.0 which is compatible to php 7.4, 8.0 and 8.1.

After installing the 8.1 DLL I got the error message on my apache err log:
"PHP Warning:  PHP Startup: imagick: Unable to initialize module\nModule compiled with module API=20210902\nPHP    compiled with module API=20220829\nThese options need to match\n in"

**Reply by lowag** [reply]
Unfortunately I can not help with this issue since I have never tried it on a Windows server. Imagick is not a main part of the Libremaint. You can try without it.

**Comment by omidhrawy** [comment]
I have the following error

gd php extension is missing
Imagick php class is missing

could you please help me?

**Reply by lowag** [reply]
You should install php gd extension and imagick class.
